modules/basegl/src/shadercomponents/lightvolumecomponent.cpp

```diff
@@ -0,0 +1,189 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2024 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/shadercomponents/lightvolumecomponent.h>
+
+#include <inviwo/core/datastructures/light/lightingconfig.h>
+#include <inviwo/core/datastructures/light/pointlight.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/util/stringconversion.h>
+#include <modules/opengl/shader/shader.h>
+#include <modules/opengl/shader/shaderutils.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/volume/volumeutils.h>
+
+#include <fmt/format.h>
+
+namespace inviwo {
+
+LightVolumeComponent::LightVolumeComponent(Processor& processor, std::string_view volumeName,
+                                           std::string_view gradientName)
+    : ShaderComponent{}
+    , lightVolume_{"lightVolume"}
+    , lightSource_{"lightSource"}
+    , scaling_{"lightVolumeScaling", "Light Volume Scaling", util::ordinalScale(1.0f, 2.0f)}
```

Comment:
Light Volume Scaling could be confused with scaling the size of the light volume
```suggestion
    , scaling_{"lightIntensityScaling", "Light Intensity Scaling", util::ordinalScale(1.0f, 2.0f)}
```